drivers/flash/flash_stm32h7x.c

```diff
@@ -60,11 +62,34 @@ static inline void _flash_stm32_sem_give(const struct device *dev)
 #define flash_stm32_sem_give(dev)
 #endif
 
+#define STM32H7_SERIES_MAX_FLASH	2048
+#define BANK2_OFFSET	(KB(STM32H7_SERIES_MAX_FLASH) / 2)
+#define REAL_FLASH_SIZE   KB(LL_GetFlashSize())
+#define SECTOR_PER_BANK  ((REAL_FLASH_SIZE / FLASH_SECTOR_SIZE) / 2)
+
+struct flash_stm32_sector {
+	int sector_index;
+	int bank;
+};
 
 bool flash_stm32_valid_range(const struct device *dev, off_t offset,
 			     uint32_t len,
 			     bool write)
 {
+#if defined(DUAL_BANK)
+	if (REAL_FLASH_SIZE < KB(STM32H7_SERIES_MAX_FLASH)) {
```

Comment:
IIUC, assumption is that in DUAL_BANK case, flash size is 2M and if total flash size is less, then banks are not continous, right ?
Maybe we should make this assumption explicit, so to ease understanding of this piece of code.

Reply:
Done with definition of #define DISCONTINUOUS_BANKS